src/autorun.ts

```diff
@@ -197,19 +197,17 @@ export const run = <T>(fn: Cb<T>): Observable<T> => new Observable(observer => {
                 .pipe(distinctUntilChanged())
                 .subscribe({
                     next: (value) => {
+                        const hadValue = v.hasValue;
                         v.hasValue = true;
                         v.value = value;
 
-                        if (isAsync) {
-                            if (v.track) {
-                                update$.next(Update.Value);
-                            } else {
-                                // NOTE: what to do if the silenced value is absent?
-                                // should we:
-                                // - interrupt computation & w scheduling re-run when first value available
-                                // - interrupt computation & w/o scheduling re-run
-                                // - continue computation w/ undefined as value of _(o)
-                            }
+                        if (isAsync && v.track) {
+                            update$.next(Update.Value);
+                        }
+                        if (!hadValue && !track) {
+                            // Untracked dep now has it's first value. So really untrack it.
+                            v.track = false;
+                            update$.next(Update.Completion);
```

Comment:
have hard time understanding this bit:
I think later we'll change naming to something like `Update.CheckCompletion`

Reply:
This was to cover the case that, after an untracked value finally had it's value, it could be that all tracked values already completed, even when the untracked value itself did not. If you remove that line, you can see what test fails :) (That's how I discovered it should be added in the first place.) I could add this explanation to the code as well...

Not sure whether it should be renamed... It's an `update$` subject which signals that something could have been updated. What is updated could be a value or the completion state.
`CheckCompletion` is an imperative name used for functions or commands but usually not for signals like this....

Reply:
Agree, we need similar name forms. I meant that `Update.Complete` sounds like something **has** completed. And it was used as such before current update. Now we use it not only in completion cases, but also to trigger that particular check: if something has completed. And thats what confused me 🤷‍♂️